Dockerfile

```diff
@@ -1,201 +1,98 @@
-  && rm SHASUMS256.txt \
-  && curl -fsSLO --compressed "https://nodejs.org/dist/latest-v18.x/$LATEST_VERSION_FILENAME.tar.xz" \
-  && curl -fsSLO --compressed "https://nodejs.org/dist/latest-v18.x/SHASUMS256.txt.asc" \
-  && gpg --batch --decrypt --output SHASUMS256.txt SHASUMS256.txt.asc \
-  && grep " $LATEST_VERSION_FILENAME.tar.xz\$" SHASUMS256.txt | sha256sum -c - \
-  && tar -xJf "$LATEST_VERSION_FILENAME.tar.xz" -C /usr/local --strip-components=1 --no-same-owner \
-  && rm "$LATEST_VERSION_FILENAME.tar.xz" SHASUMS256.txt.asc SHASUMS256.txt \
-  && ln -s /usr/local/bin/node /usr/local/bin/nodejs \
-  # smoke tests
-  && node --version \
-  && npm --version \
-  && test ! $(which python3.9)
-
-SHELL [ "/bin/bash", "-c" ]
-ENV LANG C.UTF-8
-
-# set workdir
-RUN mkdir -p /opt/code/localstack
-RUN mkdir /opt/code/localstack/localstack-core
-WORKDIR /opt/code/localstack/
-
-# create localstack user and filesystem hierarchy, perform some permission fixes
-RUN chmod 777 . && \
-    useradd -ms /bin/bash localstack && \
-    mkdir -p /var/lib/localstack && \
-    chmod -R 777 /var/lib/localstack && \
-    mkdir -p /usr/lib/localstack && \
-    mkdir /tmp/localstack && \
-    chmod -R 777 /tmp/localstack && \
-    touch /tmp/localstack/.marker && \
-    mkdir -p /.npm && \
-    chmod 755 /root && \
-    chmod -R 777 /.npm
-
-# install basic (global) tools to final image
-RUN --mount=type=cache,target=/root/.cache \
-    pip install --no-cache-dir --upgrade virtualenv
-
-# install the entrypoint script
-ADD bin/docker-entrypoint.sh /usr/local/bin/
-# add the shipped hosts file to prevent performance degredation in windows container mode on windows
-# (where hosts file is not mounted) See https://github.com/localstack/localstack/issues/5178
-ADD bin/hosts /etc/hosts
-
-# expose default environment
-# Set edge bind host so localstack can be reached by other containers
-# set library path and default LocalStack hostname
-ENV LD_LIBRARY_PATH=$JAVA_HOME/lib:$JAVA_HOME/lib/server
-ENV USER=localstack
-ENV PYTHONUNBUFFERED=1
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+        ca-certificates curl gnupg git make openssl tar pixz zip unzip groff-base iputils-ping \
+        && apt-get upgrade -y libexpat1 \
+    && apt-get clean && rm -rf /var/lib/apt/lists/*
+
+# Install Node.js without unnecessary dependencies
+RUN dpkgArch="$(dpkg --print-architecture)" && \
+    case "${dpkgArch##*-}" in \
+        amd64) ARCH='x64';; \
+        arm64) ARCH='arm64';; \
+        *) echo "unsupported architecture"; exit 1 ;; \
+    esac && \
+    curl -fsSL https://deb.nodesource.com/setup_18.x | bash - && \
+    apt-get install -y nodejs && \
+    npm cache clean --force
+
+# Set workdir and create localstack user
+WORKDIR /opt/code/localstack
+RUN useradd -ms /bin/bash localstack && \
+    mkdir -p /var/lib/localstack /tmp/localstack && \
+    chmod 777 /var/lib/localstack /tmp/localstack && \
+    touch /tmp/localstack/.marker
```

Comment:
This change here seems way cleaner, but it also does not perform some operations (chmod on `/root` and handling `/.npm`). They might be obsolete by now though?